pandas/_libs/hashtable_class_helper.pxi.in

```diff
@@ -814,7 +852,10 @@ cdef class PyObjectHashTable(HashTable):
 
         return np.asarray(locs)
 
-    def unique(self, ndarray[object] values):
+    @cython.boundscheck(False)
+    @cython.wraparound(False)
+    def _unique_no_inverse(self, ndarray[object] values):
```

Comment:
i am bothered by the naming, call these just ``_unique``, you can leave ``_unique_with_inverse``, though add a comment that makes it clear this is less performant as its doing more ops. bonus for doc-strings on these.